README.md

```diff
@@ -3,13 +3,52 @@
+  n=1,000,000<br>
+  Algo $A_2$: $3(1,000,000^3)+2(1,000,000^2)+50(1,000,000)+100$ = $3.000002*10^{18}$<br>
+  Algo $B_2$: $100(1,000,000^2)+50(1,000,000)+ 100$ = $1.00000005 * 10^{14}$<br><br>
+  Lower order terms and constants:<br>
+  n=100 <br>
+  Algo $A_1$: $2(100^2)+50(100)+100$ = 25100 or .8297% of Algo $A_1$<br>
+  Algo $B_1$: $50(100)+100$ = 5100 or .5074% of Algo $B_1$ <br><br>
+  n=1,000,000<br>
+  Algo $A_2$: $2(1,000,000^2)+50(1.000,000)+100$ = $2.00005 * 10^{12}$ or .0000667% of Algo $A_2$ <br>
+  Algo $B_2$: $50(1,000,000)+100$ = $50000100$ or .00005% of Algo $B_2$<br><br>
+  You can see from the percentages that as the input size grows the lower order terms and constants have a much smaller impact on the amount of operations the algorithm does. So in practice you need to know how impactful your lower order terms and constants will be based on individual circumstances. Yes, for large datasets the asymptotic complexity is what's going to dictate which algorithm is better. But if the dataset is under that threshold like I've shown, that can be misleading between practice and theory
+
+This also goes into seeing a algorithm with a smaller time complexity but a large constant going slower then an algorithm with a worse time complexity and smaller constant. 
+
+ Hardware as an issue. Asymptotic analysis does not take into account for hardware specifics like your cache and or memory hierarchy. Running an algorithm and getting a certain Big-O bound on one machine could be significantly different on a different machine.
+    
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+Having $O(log_2n)$ as the asymptotic complexity for searching for a specific element in a binary search tree of 1000 elements in 5 seconds has the same logarithmic growth for 10,000 elements. So, $ 5 secs / $log_2(1000)$ = $x / log_2(10,000)$ = $5 * log_2(10,000) / log_2(1000)$. 5*13.29= 66.44  $log_2(1000)$ = 9.97. (66.44/9.97) = 6.7 secs to find the same element in the tree with 10,000 items.
+
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
+  
+  1. There could be an issue with memory so the computer is running out of memory. Having fragmented data because the memory is being swapped or paged to a disk drive if there isn't enough RAM for the larger data set
+  2. There could be strings mixed in with integers. Strings are going to take longer to process. Because you're potentially going through ten times the amount of elements the ratio of strings and integers being processed could be higher, resulting in more process time for more strings.
```

Comment:
How many more elements are you potentially checking in the larger tree? It's not 10x more.

Reply:
Yes they're different the 2nd answer builds off the first reason. The first point is about the impact of constants at small input sizes and the second is about the balance of time complexity and constant factors. Both are important in theory and practice.

Reply:
No it's not 10x more because were searching logarithmically. Not linearly.  Even tho it's not 10x more it can still effect the processing in the way i've explained.

Reply:
I don't see how this could explain a difference as large as the observed one then.

Reply:
I guess I'm confused then. If there is an issue with the memory while running the algorithm and the data starts being swapped from RAM to disk space it would obviously take longer. Just because the growth isn't linear doesn't mean it's not what caused the increase in time. The memory footprint is larger then before. This could cause the issues in the increased time to search. Back to my previous reason of the computer making the wrong decisions could also be caused by a memory leak that is using up RAM space that it otherwise shouldn't . Different machines, inconsistent elements (comparing strings and ints), Memory issues where there is an issue of too much overhead causing the algorithm to swap memory between RAM and disk memory. So, what's the answer?

Reply:
Your first and last reasons are fine, it's the second I don't understand. Mixing strings and numbers etc would also affect the smaller input size, and there are not that many more elements to compare to.

Reply:
Ok that makes sense.

Reply:
I've gone back to an unbalanced tree counting linearly but due to the increased size of elements being searched there are deeper pockets to search resulting in the increased time as outside factors.